File: frontend/src/types/saved-view.ts
/**
 * Saved View Types - 保存视图类型定义
 */

export interface SavedView {
  id: number
  name: string
  description?: string
  view_type: 'list' | 'kanban'
  filters: Record<string, any>
  sort_config: SortConfigEntry[]
  columns: string[]
  group_by?: string
  is_default: boolean
  is_shared: boolean
  owner_id: number
  project_id: number
  created_at: string
  updated_at: string
}

export interface SortConfigEntry {
  field: string
  dir: 'asc' | 'desc'
}

export interface SavedViewCreate {
  name: string
  description?: string
  view_type?: 'list' | 'kanban'
  filters?: Record<string, any>
  sort_config?: SortConfigEntry[]
  columns?: string[]
  group_by?: string
  is_shared?: boolean
}

export interface SavedViewUpdate {
  name?: string
  description?: string
  view_type?: 'list' | 'kanban'
  filters?: Record<string, any>
  sort_config?: SortConfigEntry[]
  columns?: string[]
  group_by?: string
  is_shared?: boolean
}

File: backend/internal/dto/response/saved_view.go
package response

import (
	"encoding/json"
	"time"
)

// SavedViewResponse is the API response for a saved view.
type SavedViewResponse struct {
	ID          uint64          `json:"id"`
	Name        string          `json:"name"`
	Description *string         `json:"description"`
	ViewType    string          `json:"view_type"`
	Filters     json.RawMessage `json:"filters"`
	RQL         string          `json:"rql"`
	SortConfig  json.RawMessage `json:"sort_config"`
	Columns     json.RawMessage `json:"columns"`
	GroupBy     *string         `json:"group_by"`
	SubGroupBy  *string         `json:"sub_group_by"`
	IsDefault   bool            `json:"is_default"`
	IsShared    bool            `json:"is_shared"`
	OwnerID     uint64          `json:"owner_id"`
	ProjectID   uint64          `json:"project_id"`
	CreatedAt   time.Time       `json:"created_at"`
	UpdatedAt   time.Time       `json:"updated_at"`
}
